src/components/TodoList.jsx

```diff
@@ -13,51 +13,121 @@ class TodoListXP extends React.Component {
 
         this.state = {
             tasks: [
-                "do one thing",
-                "another one",
-                "last one to do"
+                { taskName: "do one thing", date: new Date() },
+                { taskName: "another one", date: new Date() },
+                { taskName: "last one to do", date: new Date() }
             ],
-            input: "nichego net"
+            input: ""
         };
-
         this.addTask = this.addTask.bind(this);
         this.removeTask = this.removeTask.bind(this);
         this.onInputChange = this.onInputChange.bind(this);
+        this.removeAllTasks = this.removeAllTasks.bind(this)
+        this.sortTasks = this.sortTasks.bind(this)
     }
 
-    removeTask(text) {
-        const { tasks } = this.state;
 
+    sortTasks(event) {
+        const { tasks } = this.state
+        const option = event.target.value
+        let newState
+        if (option === "Name asc") {
+            newState = {
+                ...this.state,
+                tasks: tasks.sort((task1, task2) => (task1.taskName.toLowerCase() > task2.taskName.toLowerCase()) ? 1 : -1)
+            }
+        }
+        if (option === "Name desc") {
+            newState = {
+                ...this.state,
+                tasks: (tasks.sort((task1, task2) => (task1.taskName.toLowerCase() > task2.taskName.toLowerCase()) ? 1 : -1)).reverse()
+            }
+        }
+        if (option === "Newer") {
+            newState = {
+                ...this.state,
+                tasks: (tasks.sort((task1, task2) => (task1.date - task2.date))).reverse()
+            }
+        }
+        if (option === "Older") {
+            newState = {
+                ...this.state,
+                tasks: tasks.sort((task1, task2) => (task1.date - task2.date))
+            }
+        }
```

Comment:
This code is not super easy to navigate. Please use switch (on string) in these situations.

Also please get rid of `let newState`. Try working only with const, using IIFE with switch inside. In more "cool" languages there is a special feature for it, called `pattern matching`, but js/ts doesn't have it, sadly :-(

e.g. in haskell you could just do this

```haskell
newTasks = case option of
    "Name asc"   -> sort tasks blablalba
    "Name desc" -> sort tasks blalbalba
    "Newer"        -> .... etc etc
```

Anyway. Notice that you **don't need** to create different STATE for each variation of `option`. Only `state.tasks` is different between those variations. So don't make newState. Make `const newTasks`, and then do `this.setState({...this.state, newTasks});`. Much more readable.